lib/import-notebook.js

```diff
@@ -144,18 +208,88 @@ function getGrammarForKernelspecName(name: string): ?atom$Grammar {
   }
 }
 
-function getCellSource(cell: Cell, commentStartString: string): string {
-  const cellType = cell.cell_type;
-  const cellMarkerKeyword = cellType === "markdown" ? "markdown" : null;
-  const cellMarker = getCellMarker(commentStartString, cellMarkerKeyword);
-  var source = cell.source;
+/**
+ * Converts notebook cells to Hydrogen code blocks.
+ *
+ * @param {Cell} cell - Notebook cell to convert
+ * @param {String} commentStartString - The comment syntax of the code language.
+ * @return {Object} - A Hydrogen Code Block.
+ */
+function toHydrogenCodeBlock(
+  cell: Cell,
+  commentStartString: string
+): {
+  cellType: HydrogenCellType,
+  code: string,
+  row: number
+} {
+  const cellType = cell.cell_type === "markdown" ? "markdown" : "codecell";
+  const cellHeader = getCellHeader(commentStartString, cellType);
+  let source = cell.source;
+  let cellLength;
   if (cellType === "markdown") {
-    source = _.map(source, line => commentStartString + line);
+    source = source.split("\n");
+    source[0] = commentStartString + source[0];
+    cellLength = source.length;
+    source = source.join(linesep + commentStartString);
+  } else {
+    cellLength = source.split("\n").length;
   }
-  return cellMarker + linesep + source.join("");
+  return {
+    cellType,
+    code: cellHeader + linesep + source,
+    row: cellLength + 1 // plus 1 for the header
+  };
 }
 
-function getCellMarker(commentStartString: string, keyword: ?string) {
-  const marker = commentStartString + "%%";
-  return keyword ? marker + ` ${keyword}` : marker;
+/**
+ * Creates a Hydrogen cell header
+ *
+ * @param {String} commentStartString - The comment syntax of the code language.
+ * @param {String} keyword - The keyword relating to the cell type.
+ * @return {String} - A Hydrogen Cell Header.
+ */
+function getCellHeader(commentStartString: string, keyword: ?string) {
+  const marker = commentStartString + "%% ";
+  return keyword ? marker + keyword : marker;
+}
+
+/**
+ * Displays previous cell results inline of the provided editor.
+ * nbCells and resultRows should be the same length.
+ *
+ * @param {atom$TextEditor} editor - The editor to display the results in.
+ * @param {Array<Cell>} nbCells - The original notebook cells.
+ * @param {Array<Number>} resultRows - The rows to display the results on.
+ */
+function importNotebookResults(
+  editor: atom$TextEditor,
+  nbCells: Array<Cell>,
+  resultRows: Array<number>
+) {
+  if (nbCells.length != resultRows.length) return;
+  let markers = store.markersMapping.get(editor.id);
+  markers = markers ? markers : store.newMarkerStore(editor.id);
+  let cellNumber = 0;
+  for (let cell of nbCells) {
+    const row = resultRows[cellNumber];
+
+    switch (cell.cell_type) {
```

Comment:
We don't need branch for `cell_type == "raw"` ?

Reply:
I will look into this.

Reply:
Well, we still don't support exporting/rendering raw cells, thus it would be okay to just import raw cells as is for now :)